scripts/lint-lockfile.js

```diff
@@ -0,0 +1,22 @@
+/**
+ * This script checks if `package-lock.json` is up-to-date with the contents of
+ * the various `package.json` files via `npm ls`.
+ *
+ * It will print the result of `npm ls` if it exits with a non-zero code;
+ * otherwise it is suppressed.
+ *
+ * It exists because it should be portable and the only other way I know how to
+ * do this is via a Bash script.
+ *
+ * @packageDocumentation
+ */
+
+const { execFileSync } = require('node:child_process')
+
+try {
+  execFileSync('npm', ['ls', '--color'], { encoding: 'utf-8' })
```

Comment:
there might be a package we could use directly instead of exec+parse. Tho that package might be `arborist`